core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -171,26 +178,57 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
 
     def abortActiveWorkflows(submissionId: UUID): Future[Seq[(Option[String], Try[ExecutionServiceStatus])]] = {
       datasource.inTransaction { dataAccess =>
-        // look up abortable WorkflowRecs for this submission
-        dataAccess.workflowQuery.findWorkflowsForAbort(submissionId).result
-      }.flatMap { workflowRecs =>
-        Future.traverse(workflowRecs) { workflowRec =>
-          Future.successful(workflowRec.externalId).zip(executionServiceCluster.abort(workflowRec, UserInfo.buildFromTokens(credential)))
+        for {
+          // look up abortable WorkflowRecs for this submission
+          wfRecs <- dataAccess.workflowQuery.findWorkflowsForAbort(submissionId).result
+          submissionRec <- dataAccess.submissionQuery.findById(submissionId).result.map(_.head)
+          submitter <- dataAccess.rawlsUserQuery.load(RawlsUserRef(RawlsUserSubjectId(submissionRec.submitterId))).map(_.get)
+          workspaceRec <- dataAccess.workspaceQuery.findByIdQuery(submissionRec.workspaceId).result.map(_.head)
+        } yield {
+          (wfRecs, submitter, workspaceRec)
         }
+      } flatMap { case (workflowRecs, submitter, workspaceRec) =>
+          for {
+            petSAJson <- samDAO.getPetServiceAccountKeyForUser(workspaceRec.namespace, submitter.userEmail)
+            petUserInfo <- googleServicesDAO.getUserInfoUsingJson(petSAJson)
+            abortResults <- Future.traverse(workflowRecs) { workflowRec =>
+              Future.successful(workflowRec.externalId).zip(executionServiceCluster.abort(workflowRec, petUserInfo))
+            }
+          } yield {
+            abortResults
+          }
+      }
+    }
+
+    def traverseWorkflowOutputs(externalWorkflowIds: Seq[WorkflowRecord], petUser: UserInfo) = {
```

Comment:
Maybe a nit?  I'm not familiar with Rawls, so without that background context I think this method could be renamed to something more descriptive.  `gatherServiceOutputs` or something like that?

Reply:
agreed, i think i banged my face on the keyboard until letters came out for this one